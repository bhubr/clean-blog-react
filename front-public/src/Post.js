import React from 'react'

class Post extends React.Component {
  render () {
    return (
      <div>
        <h2>Développement avec React</h2>
        <div>
          <p>React est devenu un incontournable du développement front-end.</p>
        </div>
      </div>
    )
  }
}

export default Post
