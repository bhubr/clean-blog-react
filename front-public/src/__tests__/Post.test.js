// Post.test.js
import React from 'react'
import Post from '../Post'
import renderer from 'react-test-renderer'

test('Check that output is correct', () => {
  // Render component
  const component = renderer.create(
    <Post title="Test post" content="Content for test post" />,
  )
  // Convert component to a tree of objects
  let tree = component.toJSON()
  // Match against snapshot
  expect(tree).toMatchSnapshot()
})
