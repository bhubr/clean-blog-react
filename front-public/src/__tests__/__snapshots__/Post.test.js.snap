// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check that output is correct 1`] = `
<div>
  <h2>
    Test post
  </h2>
  <div>
    <p>
      Content for test post
    </p>
  </div>
</div>
`;
